HELP: Added explanation what ScummVM is and where to get help for Android and iOS

sev- opened the pull request:
I want to avoid mistakes with the Android port, where many users were lost, so I am trying to squeeze in a last-minute change to the help.

Please review it and tell me what you think. I would really like to squeeze it into 2.8.0 in both ports.

The help is not yet translatable due to fonts, so it is not a problem that we will get no translations update.

If the porters agree, then it could get as a subtag only for Android and iOS, perhaps 2.8.0.1, or distributed as a patch (of course, it then be backported to branch-2-8 and branch-2-8-0).

mduggan commented on backends/platform/android/android.cpp:
```diff
@@ -1028,6 +1028,18 @@ void *OSystem_Android::getOpenGLProcAddress(const char *name) const {
+"\n"
+"We also have [online documentation for Adnroid](https://docs.scummvm.org/en/latest/other_platforms/android.html). There are [support forums](https://forums.scummvm.org/viewforum.php?f=17). On top of it, we run a Discord server with [Android support channel](https://discord.com/channels/581224060529148060/1135579923185139862).\n"
+"\n"
+"Also, please bear in mind, that many games that we support were _designed_ to be complex, sometimes very complex. So, if you are stuck in the story, consider looking for a game walktrhough.\n"
```
Small typo "walktrhough" -> "walkthrough"

bluegr replied:
```suggestion
"Also, please bear in mind, that many games that we support were _designed_ to be complex, sometimes very complex. So, if you are stuck in the story, consider looking for a game walkthrough.\n"
```

mduggan commented on gui/helpdialog.cpp:
```diff
@@ -38,6 +38,14 @@ static const char * const helpTabs[] = {
+"\n"
+"ScummVM cannot run arbitrary games, it is not an emulator of DOS, Windows or some console. It must implement all the game logic for every one of them.\n"
+"\n"
+"ScummVM is developed by a team of volunteers and is free software. We do not have an army of testers, own only a very small subset of devices, and cannot do everything. We also do not run advertisements or sell you anything. Please be mindful of this when you submit a complain or a bug report.\n"
```
"submit a complain" probably should be "submit a complaint".

bluegr replied:
```suggestion
"ScummVM is developed by a team of volunteers and is free software. We do not have an army of testers, own only a very small subset of devices, and cannot do everything. We also do not run advertisements or sell you anything. Please be mindful of this when you submit a complaint or a bug report.\n"
```

bluegr commented on gui/helpdialog.cpp:
```diff
@@ -38,6 +38,14 @@ static const char * const helpTabs[] = {
 _s(
+"## What is ScummVM\n"
+"\n"
+"ScummVM is a set of reimplemented game engines. It requires the original game data to be uploaded to the device. It takes all the graphics, sound, and other assets just as the original did and gives you the ability to play your game.\n"
```
```suggestion
"ScummVM is a set of reimplemented game engines. It requires the original game data to be uploaded to the device. It uses all the graphics, sound, and other assets just as the original did and gives you the ability to play your game.\n"
```

bluegr commented on backends/platform/ios7/ios7_osys_misc.mm:
```diff
@@ -246,6 +246,18 @@ static inline void execute_on_main_thread_async(void (^block)(void)) {
+"\n"
+"We also have [online documentation for iOS](https://docs.scummvm.org/en/latest/other_platforms/ios.html). There are [support forums](https://forums.scummvm.org/viewforum.php?f=15). On top of it, we run a Discord server with [iOS support channel](https://discord.com/channels/581224060529148060/1149456560922316911).\n"
+"\n"
+"Also, please bear in mind, that many games that we support were _designed_ to be complex, sometimes very complex. So, if you are stuck in the story, consider looking for a game walktrhough.\n"
```
```suggestion
"Also, please bear in mind, that many games that we support were _designed_ to be complex, sometimes very complex. So, if you are stuck in the story, consider looking for a game walkthrough.\n"
```

fusefib commented:
Other typos: Adnroid in [online documentation for ...]

Complete, um, suggestions for Android / iOS / What is:
```
## Help, I'm lost!

First, make sure you have the games and necessary game files ready. Check the **Where to Get the Games** section under the **General** tab. Once obtained, follow the steps outlined in the **Adding Games** tab to finish adding them on this device. Take a moment to review this process carefully, as some users encountered challenges here owing to recent Android changes.

Need more help? Refer to our [online documentation for Android](https://docs.scummvm.org/en/latest/other_platforms/android.html). Got questions? Swing by our [support forums](https://forums.scummvm.org/viewforum.php?f=17) or hop on our Discord server, which includes an [Android support channel](https://discord.com/channels/581224060529148060/1135579923185139862).

Oh, and heads up, many of our supported games are intentionally tricky, sometimes mind-bogglingly so. If you're stuck in a game, think about checking out a game walkthrough. Good luck! 
```
```
## Help, I'm lost!

First, make sure you have the games and necessary game files ready. Check the **Where to Get the Games** section under the **General** tab. Once obtained, follow the steps outlined in the **Adding Games** tab to finish adding them on this device.

Need more help? Refer to our [online documentation for iOS](https://docs.scummvm.org/en/latest/other_platforms/ios.html). Got questions? Swing by our [support forums](https://forums.scummvm.org/viewforum.php?f=17) or hop on our Discord server, which includes an [iOS support channel](https://discord.com/channels/581224060529148060/1149456560922316911).

Oh, and heads up, many of our supported games are intentionally tricky, sometimes mind-bogglingly so. If you're stuck in a game, think about checking out a game walkthrough. Good luck! 
```
```
## ScummVM at a Glance

ScummVM is a modern reinterpretation of various game engines. Once you transfer the original game data to your device, it endeavors to use it to faithfully recreate the original gaming experience. 

ScummVM isn't your typical emulator of DOS, Windows, or some. Rather than a one-size-fits-all approach, it takes a meticulous route, implementing the precise game logic for each specific title or engine it supports. ScummVM will not work with game engines it does not support.

ScummVM is developed by a team of volunteers and is free software. We lack an extensive testing team, possess only a limited range of devices, and cannot always address every request. We also do not run advertisements or sell you anything. Please be mindful of this when you submit a complaint or a bug report.

```
Preview:

## Help, I'm lost!

First, make sure you have the games and necessary game files ready. Check the **Where to Get the Games** section under the **General** tab. Once obtained, follow the steps outlined in the **Adding Games** tab to finish adding them on this device. Take a moment to review this process carefully, as some users encountered challenges here owing to recent Android changes.

Need more help? Refer to our [online documentation for Android](https://docs.scummvm.org/en/latest/other_platforms/android.html). Got questions? Swing by our [support forums](https://forums.scummvm.org/viewforum.php?f=17) or hop on our Discord server, which includes an [Android support channel](https://discord.com/channels/581224060529148060/1135579923185139862).

Oh, and heads up, many of our supported games are intentionally tricky, sometimes mind-bogglingly so. If you're stuck in a game, think about checking out a game walkthrough. Good luck! 
## Help, I'm lost!

First, make sure you have the games and necessary game files ready. Check the **Where to Get the Games** section under the **General** tab. Once obtained, follow the steps outlined in the **Adding Games** tab to finish adding them on this device.

Need more help? Refer to our [online documentation for iOS](https://docs.scummvm.org/en/latest/other_platforms/ios.html). Got questions? Swing by our [support forums](https://forums.scummvm.org/viewforum.php?f=17) or hop on our Discord server, which includes an [iOS support channel](https://discord.com/channels/581224060529148060/1149456560922316911).

Oh, and heads up, many of our supported games are intentionally tricky, sometimes mind-bogglingly so. If you're stuck in a game, think about checking out a game walkthrough. Good luck! 

## ScummVM at a Glance

ScummVM is a modern reinterpretation of various game engines. Once you transfer the original game data to your device, it endeavors to use it to faithfully recreate the original gaming experience. 

ScummVM isn't your typical emulator of DOS, Windows, or some. Rather than a one-size-fits-all approach, it takes a meticulous route, implementing the precise game logic for each specific title or engine it supports. ScummVM will not work with game engines it does not support.

ScummVM is developed by a team of volunteers and is free software. We lack an extensive testing team, possess only a limited range of devices, and cannot always address every request. We also do not run advertisements or sell you anything. Please be mindful of this when you submit a complaint or a bug report.

eriktorbjorn commented:
The commit message should say "Add explanation", not "Added explanation", I believe.

lephilousophe reviewed (commented):
The texts look good but I have two notes:
- I don't expect users to read this more than other messages we already have,
- I don't think we should rush this in the current release: it's not like it will be a game changer in the way users interact with us.

fusefib commented:
Also: My familiarity with Discord in general or Discord server administration in particular is very limited, but I believe the Discord invite link should be included (or its presence on the main page of the ScummVM website referred to), else the channel link only works for people already invited and present on the server? I'm not sure how it works on Android/iOS though.

sev- commented:
Tremendous response, team, and @fusefib, thank you for your help as well, and nice wrapping up.

Fair enough, I will not push it to the 2.8.0 release.

I updated the messages. Thank you for proofreading and added the Discord link as well.

fusefib commented on backends/platform/ios7/ios7_osys_misc.mm:
```diff
@@ -246,6 +246,18 @@ static inline void execute_on_main_thread_async(void (^block)(void)) {
+"\n"
+"First, make sure you have the games and necessary game files ready. Check the **Where to Get the Games** section under the **General** tab. Once obtained, follow the steps outlined in the **Adding Games** tab to finish adding them on this device.\n"
+"\n"
+"Need more help? Refer to our [online documentation for iOS](https://docs.scummvm.org/en/latest/other_platforms/ios.html). Got questions? Swing by our [support forums](https://forums.scummvm.org/viewforum.php?f=17) or hop on our [Discord server](https://discord.gg/4cDsMNtcpG), which includes an [iOS support channel](https://discord.com/channels/581224060529148060/1149456560922316911).\n"
```
``` suggestion
"Need more help? Refer to our [online documentation for iOS](https://docs.scummvm.org/en/latest/other_platforms/ios.html). Got questions? Swing by our [support forums](https://forums.scummvm.org/viewforum.php?f=15) or hop on our [Discord server](https://discord.gg/4cDsMNtcpG), which includes an [iOS support channel](https://discord.com/channels/581224060529148060/1149456560922316911).\n"
```
Apologies, I think I introduced an error, the link here should be:
https://forums.scummvm.org/viewforum.php?f=15
not 
https://forums.scummvm.org/viewforum.php?f=17
